commit b718594e7acfaa9a18c5486bf3306d603fc89585 (refs/changes/81/3981/2)
Author: Pedro Palazon Candel <pedro@joyent.com>
Date:   2018-05-22T12:35:50+02:00 (1 year, 5 months ago)
    
    TOOLS-1517 scdcam check-config should not complain about missing docker/cns vars in SAPI
    Reviewed by: Marsell Kukuljevic <marsell@joyent.com>
    Approved by: Marsell Kukuljevic <marsell@joyent.com>

diff --git a/lib/cli/do_check_config.js b/lib/cli/do_check_config.js
index 577ed3a..b1ad8b5 100644
--- a/lib/cli/do_check_config.js
+++ b/lib/cli/do_check_config.js
@@ -298,6 +298,10 @@ CheckConfig.prototype.execute = function cExecute(opts, args, cb) {
             // remove most of these variables anyway, and left a single
             // value for *_pw.
             services.forEach(function checkSvcCfgVars(s) {
+                if (s === 'cns' || s === 'portolan' || s === 'docker' ||
+                    s === 'cmon' || s === 'volapi') {
+                    return;
+                }
                 if (!sdc[s + '_root_pw'] && s !== 'manta' && s !== 'sapi') {
                     errs.push(sprintf('Missing %s_root_pw in SAPI', s));
                 }
diff --git a/test/check-config.test.js b/test/check-config.test.js
index 64bfd05..5e56c1e 100644
--- a/test/check-config.test.js
+++ b/test/check-config.test.js
@@ -25,11 +25,7 @@ test('sdcadm check-config --help', function (t) {
 });
 
 
-// Right now check-config is pretty much useless. Need to update it and take
-// a complete review of a proper test suite.
-// Need to fix https://jira.joyent.us/browse/TOOLS-1517 in order to avoid the
-// test failing.
-test.skip('sdcadm check-config', function (t) {
+test('sdcadm check-config', function (t) {
     exec('sdcadm check-config', function (err, stdout, stderr) {
         t.ifError(err);
 
